src/freenet/pluginmanager/PluginInfoWrapper.java

```diff
@@ -58,7 +61,9 @@ public PluginInfoWrapper(Node node, FredPlugin plug, String filename, boolean is
 		isIPDetectorPlugin = (plug instanceof FredPluginIPDetector);
 		isPortForwardPlugin = (plug instanceof FredPluginPortForward);
 		isMultiplePlugin = (plug instanceof FredPluginMultiple);
-		isFCPPlugin = (plug instanceof FredPluginFCP);
+        isFCPPlugin = (plug instanceof FredPluginFCP);
+        isFCPServerPlugin
+            = (plug instanceof FredPluginFCPMessageHandler.ServerSideFCPMessageHandler);
```

Comment:
I've added what you said as "TODO: Code quality: " in commit 2ebdd5610c36db91f95215aaed8771f81bab2526 (while specifying you as commit author).
I would suggest that I do keep the code which I added the way it is, so it is coherent with the existing code while the existing code is the way it is / so that I don't have to start converting the whole of it to keep things coherent.

ACK?